aloe/registry.py

```diff
@@ -214,8 +214,12 @@ def load(self, sentence, func):
         self.steps[step_re.pattern] = (step_re, func)
 
         try:
+            if not getattr(func, 'patterns', None):
+                func.patterns = set()
+
+            func.patterns.add(step_re.pattern)
             func.sentence = sentence
-            func.unregister = partial(self.unload, step_re.pattern)
+            func.unregister = partial(self.unload, *func.patterns)
```

Comment:
Good! Is there a test for this? Without the extended steps, just registering a function twice and calling `unregister` should unregister both.

However, where are `patterns` used?